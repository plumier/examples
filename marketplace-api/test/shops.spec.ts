import supertest from "supertest"
import createApp from "../src/app"
import { closeConnection, createUser, ignore, userToken } from "./_helper"


afterEach(async () => {
    await closeConnection()
})

describe("Shop", () => {
    it("Should not able to create shop without authenticated", async () => {
        const app = await createApp({ mode: "production" })
        await supertest(app.callback())
            .post("/api/v1/shops")
            .send({ name: "Putra Mahkota" })
            .expect(403)
    })
    it("Should able to register for any login user", async () => {
        const app = await createApp({ mode: "production" })
        const user = await createUser(app)
        await supertest(app.callback())
            .post("/api/v1/shops")
            .send({ name: "Putra Mahkota" })
            .set("Authorization", `Bearer ${user.token}`)
            .expect(200)
    })
    it("Should assign person who register as owner", async () => {
        const app = await createApp({ mode: "production" })
        const user = await createUser(app)
        const { body } = await supertest(app.callback())
            .post("/api/v1/shops")
            .send({ name: "Putra Mahkota" })
            .set("Authorization", `Bearer ${user.token}`)
            .expect(200)
        const { body: result } = await supertest(app.callback())
            .get(`/api/v1/shops/${body.id}/users`)
            .set("Authorization", `Bearer ${user.token}`)
            .expect(200)
        expect(result[0]).toMatchSnapshot({ userId: expect.any(Number) })
    })
    it("Should restrict other user to modify other shop", async () => {
        const app = await createApp({ mode: "production" })
        const user = await createUser(app)
        const { body } = await supertest(app.callback())
            .post("/api/v1/shops")
            .send({ name: "Putra Mahkota" })
            .set("Authorization", `Bearer ${user.token}`)
            .expect(200)
        await supertest(app.callback())
            .patch(`/api/v1/shops/${body.id}`)
            .set("Authorization", `Bearer ${userToken}`)
            .expect(401)
    })
})