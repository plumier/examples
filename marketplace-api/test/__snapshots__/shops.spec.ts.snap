// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop Should assign person who register as owner 1`] = `
Object {
  "name": "John Doe",
  "role": "ShopOwner",
  "userId": Any<Number>,
}
`;
